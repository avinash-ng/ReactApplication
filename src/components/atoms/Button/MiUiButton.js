import React from 'react'
import { Button } from "@material-ui/core"
const MiUiButton = props => {
    return (
        <Button className={props.className} style={props.style} variant={props.variant} onClick={props.onClick}>{props.buttonName}</Button> 
    )
}
export default MiUiButton