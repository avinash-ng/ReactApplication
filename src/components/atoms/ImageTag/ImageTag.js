import React from 'react'
import './ImageTag.css'
const ImageTag = props => {
    return (
        <h3 className={props.className} style={{ width: '450px', marginBottom: '20px' }} onClick={props.onClicked} >{props.object.name}</h3>
    )
}
export default ImageTag