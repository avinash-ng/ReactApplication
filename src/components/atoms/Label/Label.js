import React from 'react'
import PropTypes from 'prop-types'
const Label = (props) => {
    return (
        <span className="a__label" >{props.label}</span>
        
    )
}

Label.propTypes = {
    label : PropTypes.string.isRequired
}
export default Label