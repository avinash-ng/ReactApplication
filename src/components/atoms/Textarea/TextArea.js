import React from 'react'
import { TextareaAutosize } from '@material-ui/core';
const TextArea = (props) => {
    return (
        <TextareaAutosize placeholder={props.placeholder} ></TextareaAutosize>
    )

}

export default TextArea;