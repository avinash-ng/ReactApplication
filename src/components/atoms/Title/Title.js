import React from 'react'
import { Typography } from '@material-ui/core'
import { useAuth0 } from '../../../react-auth0-spa'

const Title = (props) => {
    
    return (
        // <h6> Hi {props.title} Welcome</h6>
        <Typography variant="h6" noWrap>
                Hi {props.name}      
        </Typography>
    )
}

export default Title;