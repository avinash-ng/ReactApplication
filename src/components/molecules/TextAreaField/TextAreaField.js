import  TextArea  from "../../atoms/Textarea/TextArea"
import Label from "../../atoms/Label/Label"
import React from 'react'
const TextAreaField = props => {
    return (
        <div>
        <Label label={props.label}></Label>
        <TextArea placeholder={props.placeholder}></TextArea>
        </div>
    )
}
export default TextAreaField;