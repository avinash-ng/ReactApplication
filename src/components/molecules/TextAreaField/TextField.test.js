import TextAreaField from './TextAreaField'
import expect from 'expect'
import sinon from 'sinon';
import { shallow, configure, mount } from 'enzyme'
import React from 'react'
import Adapter from 'enzyme-adapter-react-16';
import { Button } from '@material-ui/core';
import ImageTag from '../../atoms/ImageTag/ImageTag'
import MiUiButton from '../../atoms/Button/MiUiButton'
import Title from '../../atoms/Title/Title';
import Label from '../../atoms/Label/Label';
import TextArea from '../../atoms/Textarea/TextArea';

configure({ adapter: new Adapter() });
describe('<TextAreaField /> ', () => {
    it('renders <Label /> element properly', () => {
        const label='Label '

        const RenderComponent = shallow(
            <TextAreaField label={label}></TextAreaField>
        );
        // console.log(RenderComponent)
        expect(RenderComponent.find(Label).exists()).toBe(true)
    })
    
    it('renders <TextAreaAutoSize /> element properly', () => {
        
        const label='Label '
        const RenderComponent = shallow(
            <TextAreaField label={label}></TextAreaField>
        );
        // console.log(RenderComponent)
        expect(RenderComponent.find(TextArea).exists()).toBe(true)
    })
    it('recieved props correctly ', () => {
        const label = "Label 1"
        const placeholder = "enter your string..."
        const RenderComponent = shallow(<TextAreaField label={label} placeholder={placeholder }></TextAreaField>)
        expect(RenderComponent.find(TextArea).prop('placeholder')).toBe(placeholder)
    })


    it('Matches snapshot', () => {
        const RenderComponent = shallow(<TextAreaField label="Label"></TextAreaField>)
        expect(RenderComponent).toMatchSnapshot()
    })
})