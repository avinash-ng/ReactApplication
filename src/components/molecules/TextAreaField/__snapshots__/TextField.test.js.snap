// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextAreaField />  Matches snapshot 1`] = `ShallowWrapper {}`;
